libs/model/src/main/kotlin/de/cyface/model/ExportOptions.kt

```diff
@@ -0,0 +1,171 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the Serialization. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.cyface.model
+
+/**
+ * The options which describe which data should be exported.
+ *
+ * @author Armin Schnabel
+ * @version 1.0.0
+ * @since 3.3.0
+ * @property format The format in which the data should be exported.
+ * @property type The type of data which should be exported.
+ * @property includeHeader `true` if the export should start with a header line, if supported by the [format].
+ * @property includeUserId `true` if the user id should be included in the export.
+ * @property includeUsername `true` if the username should be included in the export.
+ */
+class ExportOptions {
+    var format: DataFormat = DataFormat.UNDEFINED
+    var type: DataType = DataType.UNDEFINED
+    var includeHeader: Boolean = false
+    var includeUserId: Boolean = false
+    var includeUsername: Boolean = false
+
+    fun format(format: DataFormat): ExportOptions {
+        this.format = format
+        return this
+    }
+
+    fun type(type: DataType): ExportOptions {
+        this.type = type
+        return this
+    }
+
+    fun includeHeader(includeHeader: Boolean): ExportOptions {
+        require(!includeHeader || format == DataFormat.CSV) { "Format without header support: $format" }
+        this.includeHeader = includeHeader
+        return this
+    }
+
+    fun includeUserId(includeUserId: Boolean): ExportOptions {
+        this.includeUserId = includeUserId
+        return this
+    }
+
+    fun includeUsername(includeUsername: Boolean): ExportOptions {
+        this.includeUsername = includeUsername
+        return this
+    }
+}
+
+/**
+ * Supported export data format.
+ *
+ * @author Armin Schnabel
+ * @version 1.0.0
+ * @since 3.3.0
+ * @param parameterValue The parameter value which represents the enum option.
+ */
+enum class DataFormat(
+    @Suppress("MemberVisibilityCanBePrivate") val parameterValue: String
+) {
+    /**
+     * Default value when no format was specified.
+     */
+    UNDEFINED("undefined"),
+
+    /**
+     * Comma separated values.
+     */
+    CSV("csv"),
+
+    /**
+     * JSON format.
+     */
+    @Suppress("unused") // Used
+    JSON("json");
+
+    companion object {
+        private val BY_PARAMETER_VALUE: MutableMap<String, DataFormat> = HashMap()
+
+        init {
+            values().forEach { format ->
+                BY_PARAMETER_VALUE[format.parameterValue] = format
+            }
+        }
+
+        /**
+         * Returns the [DataFormat] from it's `#parameterValue` value.
+         *
+         * @param parameterValue The `String` value of the parameterValue.
+         * @return The `DataFormat` for the parameterValue.
+         */
+        @Suppress("unused") // Part of the API
+        fun valueOfParameterValue(parameterValue: String): DataFormat? {
```

Comment:
I would just call this method `of(parameterValue: String): DataFormat?`. The call would look like `DataFormat.of("json")`. This avoids having 'value' three times in the signature and 'parameterValue' twice.